ksql-functional-tests/src/test/resources/query-validation-tests/sum.json

```diff
@@ -191,6 +191,40 @@
         {"topic": "S2", "key": 0, "value": {"ID":0,"SUM_VAL":4.0}},
         {"topic": "S2", "key": 0, "value": {"ID":0,"SUM_VAL":5.0}}
       ]
+    },
+    {
+      "name": "sum list of doubles into a single double",
```

Comment:
Can we add tests for `BIGINT` and `DECIMAL` please?

Also, can you add a `null` value in each test please?

Reply:
Actually, there is an issue with BigDecimal data types. It is not supported correctly currently. When a UDAF is defined with BigDecimal as input argument, the schema associated with it is `Decimal(1,0)`, and that's how it is added to the `UdfIndex`. However, when an aggregate function is called on a column with data type `Decimal(4,2)` for instance, the code tries to find the method with this input type and cannot find it (as it was added using the default `decimal(1,0)`. Looking at the test cases, this has not been checked for in other UDAFs.
I will need to talk with @agavra about this. I will remove support for BigDecimal for now.